File: app/src/main/java/com/kelsos/mbrc/presenters/MiniControlPresenter.java
package com.kelsos.mbrc.presenters;

import com.kelsos.mbrc.model.MainDataModel;
import com.kelsos.mbrc.views.MiniControlView;
import javax.inject.Inject;

public class MiniControlPresenter extends BasePresenter<MiniControlView> {

  @Inject MainDataModel model;

  public MiniControlPresenter() {

  }

  public void load() {
    if (!isAttached()) {
      return;
    }

    getView().updateCover(model.getCover());
    getView().updateState(model.getPlayState());
    getView().updateTrackInfo(model.getTrackInfo());
  }

}

File: app/src/main/kotlin/com/kelsos/mbrc/presenters/MiniControlPresenter.kt
package com.kelsos.mbrc.presenters

import com.kelsos.mbrc.constants.Const
import com.kelsos.mbrc.model.MainDataModel
import com.kelsos.mbrc.mvp.BasePresenter
import com.kelsos.mbrc.views.MiniControlView
import javax.inject.Inject

class MiniControlPresenter : BasePresenter<MiniControlView>() {

  @Inject lateinit var model: MainDataModel

  fun load() {
    if (!isAttached) {
      return
    }

    view?.updateCover(Const.COVER_FILE)
    view?.updateState(model.playState)
    view?.updateTrackInfo(model.trackInfo)
  }

}
